### PR TITLE
Device in transformer model recommend

### PR DESCRIPTION
**Pull request by spirinamayya**
Added recommend ranking kwargs

### CODE REVIEW

File: rectools/models/sasrec.py [hunk 1]
```diff
@@ -1137,14 +1137,16 @@ def __init__(  # pylint: disable=too-many-arguments, too-many-locals
         epochs: int = 3,
         verbose: int = 0,
         deterministic: bool = False,
-        cpu_n_threads: int = 0,
+        recommend_device: str = "cpu",
```
Comment (blondered) [inline rectools/models/sasrec.py:1140]: default value should be "auto".
And type should be exactly the one that is accepted in trainer accelerator, not string

File: rectools/models/sasrec.py [hunk 2]
```diff
@@ -1137,14 +1137,16 @@ def __init__(  # pylint: disable=too-many-arguments, too-many-locals
...
+        recommend_cpu_n_threads: int = 0,
         trainer: tp.Optional[Trainer] = None,
         item_net_block_types: tp.Sequence[tp.Type[ItemNetBase]] = (IdEmbeddingsItemNet, CatFeaturesItemNet),
         pos_encoding_type: tp.Type[PositionalEncodingBase] = LearnableInversePositionalEncoding,
         lightning_module_type: tp.Type[SessionEncoderLightningModuleBase] = SessionEncoderLightningModule,
     ) -> None:
         super().__init__(verbose)
-        self.n_threads = cpu_n_threads
+        self.n_recommend_threads = recommend_cpu_n_threads
```
[the remark this answers is not in the document]
Comment (spirinamayya) [reply rectools/models/sasrec.py]: fixed

File: rectools/models/sasrec.py [hunk 3]
```diff
@@ -1198,8 +1200,8 @@ def _fit(
             n_item_extra_tokens=n_item_extra_tokens,
         )
 
-        self.trainer = deepcopy(self._trainer)
-        self.trainer.fit(self.lightning_model, train_dataloader)
+        trainer = deepcopy(self._trainer)
```
[the remark this answers is not in the document]
Comment (spirinamayya) [reply rectools/models/sasrec.py]: fixed

File: rectools/models/sasrec.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (spirinamayya) [reply rectools/models/sasrec.py]: fixed

Comment (spirinamayya) [reply rectools/models/sasrec.py]: changed to "auto"

File: rectools/models/sasrec.py [hunk 4]
```diff
@@ -1293,7 +1300,69 @@ def torch_model(self) -> TransformerBasedSessionEncoder:
...
+        Learning rate.
+    epochs: int, default 3
+        Number of training epochs.
+    verbose: int, default 0
+        Verbosity level.
+    deterministic: bool, default ``False``
+        If ``True``, sets deterministic algorithms for PyTorch operations.
+        Use `pytorch_lightning.seed_everything` together with this parameter to fix the random state.
+    recommend_device: str, default "auto"
+        Device for recommend. Used at predict_step of lightning module.
```
Comment (blondered) [inline rectools/models/sasrec.py:1344]: Please add this to all 3 arguments with the same logic:
```
If you want to change this parameter after model is initialized,
you can manually assign new value to model `recommend_<param>` attribute.
```

File: rectools/models/sasrec.py [hunk 5]
```diff
@@ -1314,7 +1383,9 @@ def __init__(  # pylint: disable=too-many-arguments, too-many-locals
         epochs: int = 3,
         verbose: int = 0,
         deterministic: bool = False,
-        cpu_n_threads: int = 0,
+        recommend_device: str = "auto",
```
Comment (blondered) [inline rectools/models/sasrec.py:1386]: Нужен везде тип аналогичный лайтнингу. Это union[str, Accelerator] https://github.com/Lightning-AI/pytorch-lightning/blob/64d0d02c50f9d9d7fc7dd135ead065b26af76a85/src/lightning/pytorch/trainer/trainer.py#L92C9-L92C55

File: rectools/models/sasrec.py [hunk 6]
```diff
@@ -1137,14 +1138,18 @@ def __init__(  # pylint: disable=too-many-arguments, too-many-locals
...
+        recommend_cpu_n_threads: int = 0,
```
[the remark this answers is not in the document]
Comment (spirinamayya) [reply rectools/models/sasrec.py]: fixed

File: rectools/models/sasrec.py [hunk 7]
```diff
@@ -1137,14 +1138,18 @@ def __init__(  # pylint: disable=too-many-arguments, too-many-locals
...
+        recommend_use_gpu_ranking: bool = False,
```
[the remark this answers is not in the document]
Comment (spirinamayya) [reply rectools/models/sasrec.py]: fixed

File: rectools/models/sasrec.py (hunk 4 above)
Comment (spirinamayya) [reply rectools/models/sasrec.py:1345]: fixed

File: rectools/models/sasrec.py (hunk 5 above)
Comment (spirinamayya) [reply rectools/models/sasrec.py]: fixed

**Pull request merged**